File: 2006Scape Server/plugins/plugin/click/item/ItemSecondClick.java
package plugin.click.item;

import com.rs2.event.EventContext;
import com.rs2.event.EventSubscriber;
import com.rs2.event.SubscribesTo;
import com.rs2.event.impl.ItemSecondClickEvent;
import com.rs2.game.players.Player;

import static com.rs2.game.content.StaticItemList.YOYO;

@SubscribesTo(ItemSecondClickEvent.class)
public final class ItemSecondClick implements EventSubscriber<ItemSecondClickEvent> {

    @Override
    public void subscribe(EventContext context, Player player, ItemSecondClickEvent event) {
        if (player.playerRights == 3) {
            player.getPacketSender().sendMessage("[ItemClick#2] - ItemId: " + event.getId());
        }

        switch (event.getId()) {

            case YOYO:
                player.startAnimation(1459);
                break;

        }
    }

}

File: 2006Scape Server/plugins/plugin/click/item/ItemSecondClick.kt
package plugin.click.item

import com.rs2.event.EventContext
import com.rs2.event.EventSubscriber
import com.rs2.event.SubscribesTo
import com.rs2.event.impl.ItemSecondClickEvent
import com.rs2.game.content.StaticItemList.YOYO
import com.rs2.game.players.Player

@SubscribesTo(ItemSecondClickEvent::class)
class ItemSecondClick : EventSubscriber<ItemSecondClickEvent> {

	override fun subscribe(context: EventContext, player: Player, event: ItemSecondClickEvent) {

		if (player.playerRights >= 3) {
			player.packetSender.sendMessage("[ItemClick#2] - ItemId: ${event.id}")
        }
		
		when(event.id) {

			YOYO -> player.startAnimation(1459)

		}
		
		
	}
	
}